xapian-applications/omega/index_file.cc

```diff
@@ -153,8 +153,19 @@ void
 index_add_default_libraries()
 {
 #if defined HAVE_POPPLER
-    index_library("application/pdf", new Worker("omindex_pdf"));
+    Worker* omindex_pdf = new Worker("omindex_pdf");
+    index_library("application/pdf", omindex_pdf);
 #endif
+#if defined HAVE_LIBEBOOK
+    Worker* omindex_ebook = new Worker("omindex_ebook");
+    index_library("application/vnd.palm", omindex_ebook);
+    index_library("application/x-fictionbook+xml", omindex_ebook);
+    index_library("application/x-zip-compressed-fb2", omindex_ebook);
+    index_library("application/x-sony-bbeb", omindex_ebook);
+    index_library("application/x-tcr-ebook", omindex_ebook);
+    index_library("application/java-archive", omindex_ebook);
```

Comment:
Should I remove `application/java-archive` or `application/x-zip-compressed-fb2` from here?

Reply:
Listing `application/x-zip-compressed-fb2` here seems OK - that will mean we index such files if libmagic identifies that type from the contents or the user passes `--mime-type-match='*.fb2.zip':application/x-zip-compressed-fb2` (we could support via that mechanism internally, but I think it makes more sense to add support for multi-level extensions).

Having a rule for `application/java-archive` seems more problematic since libmagic will likely report that for many files which aren't QiOO format (because it's the type for Java JAR files).  Probably the best way to handle this is to make up a mime type for this format in the `x-` space which is reserved for private use (e.g. `application/x-qioo`) and use that here.  Then a user with such files can use `--mime-type=jar:application/x-qioo` to enable indexing of them.

Reply:
Okay, I am going to change it.